apps/remote_control/test/lexical/remote_control/code_intelligence/definition_test.exs

```diff
@@ -0,0 +1,318 @@
+  import Lexical.Test.CodeSigil
+  import Lexical.Test.CursorSupport
+  import Messages
+
+  use ExUnit.Case, async: false
+
+  defp with_navigation_project(_ctx) do
+    project = project(:navigations)
+    {:ok, _, _} = RemoteControl.start_link(project, self())
+
+    on_exit(fn ->
+      :ok = RemoteControl.stop(project)
+    end)
+
+    RemoteControl.Api.schedule_compile(project, true)
+    assert_receive project_compiled(), 5000
+
+    %{project: project}
+  end
+
+  defp with_referenced_file(%{project: project}, file \\ "my_definition.ex") do
+    path = file_path(project, Path.join("lib", file))
+    %{uri: SourceFile.Path.ensure_uri(path)}
+  end
+
+  defp uses_file_uri(project) do
+    project
+    |> file_path(Path.join("lib", "my_module.ex"))
+    |> SourceFile.Path.ensure_uri()
+  end
+
+  defp open_uses_file(project, content) do
+    uri = uses_file_uri(project)
+
+    with :ok <- RemoteControl.call(project, SourceFile.Store, :open, [uri, content, 1]) do
+      RemoteControl.call(project, SourceFile.Store, :fetch, [uri])
+    end
+  end
+
+  setup_all ~w(with_navigation_project)a
+
+  setup %{project: project} do
+    uri = uses_file_uri(project)
+
+    # NOTE: We need to make sure every tests start with fresh caller content file
+    on_exit(fn ->
+      :ok = RemoteControl.call(project, SourceFile.Store, :close, [uri])
+    end)
+  end
+
+  describe "definition/2 when making remote call by alias" do
+    setup [:with_referenced_file]
+
+    test "find the definition of a remote function call", %{project: project, uri: referenced_uri} do
+      uses_content = ~q[
+        defmodule UsesRemoteFunction do
+          alias MyDefinition
+
+          def uses_greet() do
+            MyDefinition.greet|("World")
+          end
+        end
+        ]
+
+      assert {:ok, ^referenced_uri, definition_line} = definition(project, uses_content)
+      assert definition_line == ~S[  def «greet»(name) do]
+    end
+
+    test "find the definition of the module", %{project: project, uri: referenced_uri} do
+      uses_content = ~q[
+        defmodule UsesRemoteFunction do
+          alias MyDefinition
+
+          def uses_greet() do
+            MyDefinition|.greet("World")
+          end
+        end
+        ]
+
+      assert {:ok, ^referenced_uri, definition_line} = definition(project, uses_content)
+      assert definition_line == ~S[defmodule «MyDefinition» do]
+    end
+
+    test "find the macro definition", %{project: project, uri: referenced_uri} do
+      uses_content = ~q[
+        defmodule UsesRemoteFunction do
+          alias MyDefinition
+
+          def uses_macro() do
+            MyDefinition.print_hello|()
+          end
+        end
+        ]
+
+      assert {:ok, ^referenced_uri, definition_line} = definition(project, uses_content)
+      assert definition_line == ~S[  defmacro «print_hello» do]
+    end
+
+    test "it can't find the right arity function definition", %{project: project} do
```

Comment:
make a comment here explaining the issue. You should consider skipping this test rather than using a TODO.
TODOs almost never get done, but an asterisk in our tests tell us that something is not quite right.